app/src/main/java/in/ac/bits_pilani/goa/ard/utils/Browser.java

```diff
@@ -0,0 +1,408 @@
+
+    /**
+     * Bundle ActionButton.
+     */
+    private Bundle actionButton;
+
+    /**
+     * Should action button be enabled.
+     */
+    private boolean actionButtonEnabled;
+
+    /**
+     * Constructor initiates parameters to default values.
+     * @param hostActivity activity from which wab tab is launched.
+     */
+    public Browser(final Activity hostActivity) {
+        this.hostActivity = hostActivity;
+
+        final int[] attrs = {R.attr.customTab_toolbar_color};
+        final TypedArray ta = hostActivity.obtainStyledAttributes(R.style.AppTheme, attrs);
+        setToolBarColor(ta.getColor(0, ActivityCompat.getColor(hostActivity, R.color.colorPrimary)));
+        ta.recycle();
+
+        setCloseButtonIcon(BitmapFactory.decodeResource(hostActivity.getResources(), R.drawable.ic_close));
+        setStartAnimation(R.anim.slide_in_up, R.anim.slide_out_down);
+        setExitAnimations(R.anim.slide_in_down, R.anim.slide_out_up);
+        enableUrlBarHiding();
+        enableTitle();
+        enableShareInMenu();
+        disableActionButton();
+    }
+
+    /**
+     * Sets toolbar color.
+     * @param color The {@link Color}
+     * @return current instance.
+     */
+    public Browser setToolBarColor(@ColorInt final int color) {
+        this.toolBarColor = color;
+        return this;
+    }
+
+    /**
+     * Sets the Close button icon for the custom tab.
+     * @param icon The icon {@link Bitmap}
+     * @return current instance.
+     */
+    public Browser setCloseButtonIcon(@NonNull final Bitmap icon) {
+        // According to docs, the height and width of icon should be 24dp.
+        closeButtonIcon = Bitmap.createScaledBitmap(
+                icon,
+                (int) dpToPx(hostActivity, CLOSE_BUTTON_WIDTH),
+                (int) dpToPx(hostActivity, CLOSE_BUTTON_HEIGHT),
+                true);
+        return this;
+    }
+
+
+    /**
+     * Sets the start animations.
+     * @param enterResId Resource ID of the "enter" animation for the browser.
+     * @param exitResId Resource ID of the "exit" animation for the application.
+     * @return current instance.
+     */
+    public Browser setStartAnimation(@AnimRes final int enterResId, @AnimRes final int exitResId) {
+        this.startAnimEnterResId = enterResId;
+        this.startAnimExitResId = exitResId;
+        return this;
+    }
+
+    /**
+     * Sets the exit animations.
+     * @param enterResId Resource ID of the "enter" animation for the application.
+     * @param exitResId Resource ID of the "exit" animation for the browser.
+     * @return current instance.
+     */
+    public Browser setExitAnimations(@AnimRes final int enterResId, @AnimRes final int exitResId) {
+        this.exitAnimEnterResId = enterResId;
+        this.exitAnimExitResId = exitResId;
+        return this;
+    }
+
+    /**
+     * Don't hide url bar when user scrolls.
+     * @return current instance.
+     */
+    public Browser disableUrlBarHiding() {
```

Comment:
for all enable/disable methods, combine them into one, **set method** like `setUrlBarHiding(boolean val)`